numba/types/misc.py

```diff
@@ -346,6 +350,66 @@ def struct(self):
     def methods(self):
         return self.class_type.methods
 
+    def is_hashable(self):
+        return '__hash__' in self.methods
+
+    def get_user_hash(self, context, sig):
+        method = self.jitmethods["__hash__"]
+        disp_type = Dispatcher(method)
+        # Get call signature, which may have different return type
+        callsig = disp_type.get_call_type(context.typing_context, sig.args, {})
+        call = context.get_function(disp_type, callsig)
+        return call, callsig
+
+    def _get_user_equality_fn(self, context, sig, method):
+        disp_type = Dispatcher(method)
+        # Get call signature, which may have different return type
+        callsig = disp_type.get_call_type(context.typing_context, sig.args, {})
+        call = context.get_function(disp_type, callsig)
+        return call, callsig
+
+    def supports_eq(self):
```

Comment:
Do we need all those mostly identical methods? Perhaps we can write `supports_operator('==')` and `get_operator('==')` instead.